bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java

```diff
@@ -158,28 +148,26 @@ private void updateChannels(ReportConfiguration config) {
     @Override
     public void dispose() {
         logger.debug("Disposing the OpenUV handler.");
-        ScheduledFuture<?> refresh = refreshJob;
-        if (refresh != null && !refresh.isCancelled()) {
-            refresh.cancel(true);
-        }
-        refreshJob = null;
 
-        ScheduledFuture<?> uxMax = uvMaxJob;
-        if (uxMax != null && !uxMax.isCancelled()) {
-            uxMax.cancel(true);
+        cancelFuture(refreshJob);
+        cancelFuture(uvMaxJob);
```

Comment:
You could `refreshJob`  and `uvMaxJob` to null just after.

Reply:
Done